doc/hardware/peripherals/dma/index.rst

```diff
@@ -0,0 +1,61 @@
+.. _dma_api:
+
+Direct Memory Access (DMA)
+##########################
+
+Overview
+********
+
+Direct Memory Access (Controller) is a commonly provided co-processor of sorts
+that can typically offload transferring data to and from peripherals and memory.
+
+The DMA API in Zephyr is not necessarily a portable API across devices but
+instead a union of common DMA features. This API can still provide a reasonable
+abstraction given some care for peripheral drivers.
```

Comment:
I feel like the wording  can be improved but feel free to ignore if this doesn't sound any better to you or if it changes the meaning too much.

```suggestion
The DMA API in Zephyr does not guarantee portability across different devices.
Nevertheless, and with appropriate consideration for peripheral drivers, it can
offer a substantial level of abstraction for common DMA operations.
```

Reply:
btw it feels like your commit message (2nd paragraph specifically) might contain wording that's better that what ended up being captured in this paragraph, so maybe have another pass?